pkg/kubeapiserver/admission/initializer.go

```diff
@@ -89,9 +129,30 @@ func NewPluginInitializer(internalClient internalclientset.Interface,
 	}
 }
 
+// SetServiceResolver sets the service resolver which is needed by some plugins.
+func (i *PluginInitializer) SetServiceResolver(s ServiceResolver) *PluginInitializer {
+	i.serviceResolver = s
+	return i
+}
+
+// SetClientCert sets the client cert & key (identity used for calling out to
+// web hooks) which is needed by some plugins.
+func (i *PluginInitializer) SetClientCert(cert, key []byte) *PluginInitializer {
+	i.clientCert = cert
+	i.clientKey = key
+	return i
+}
+
+// SetWebhookSource sets the webhook source-- admittedly this is probably
+// specific to the external admission hook plugin.
+func (i *PluginInitializer) SetWebhookSource(w WebhookSource) *PluginInitializer {
```

Comment:
I'm going to remove SetWebhookSource. This should be a SetExternalClient, and we should initialize a webhookconfiguration manager in the method, like what Clayton did in https://github.com/kubernetes/kubernetes/pull/46672/commits/b75238f4e5351e0a68e679597cff7342d86fc92e#diff-3551f205184f7273b78e802da8fdc72fR93.